crates/precompile/src/bls12_381/g1_msm.rs

```diff
@@ -39,26 +40,24 @@ pub(super) fn g1_msm(input: &Bytes, gas_limit: u64) -> PrecompileResult {
         return Err(PrecompileError::OutOfGas);
     }
 
-    let mut g1_points: Vec<blst_p1> = Vec::with_capacity(k);
+    let mut g1_points: Vec<blst_p1_affine> = Vec::with_capacity(k);
     let mut scalars: Vec<u8> = Vec::with_capacity(k * SCALAR_LENGTH);
     for i in 0..k {
-        let slice = &input[i * G1_MSM_INPUT_LENGTH..i * G1_MSM_INPUT_LENGTH + G1_INPUT_ITEM_LENGTH];
+        let encoded_g1_element =
+            &input[i * G1_MSM_INPUT_LENGTH..i * G1_MSM_INPUT_LENGTH + G1_INPUT_ITEM_LENGTH];
 
-        // BLST batch API for p1_affines blows up when you pass it a point at infinity, so we must
-        // filter points at infinity (and their corresponding scalars) from the input.
-        if slice.iter().all(|i| *i == 0) {
+        // Filter out the point at infinity from the MSM for two reasons:
+        // - The current library `blst` does not handle it gracefully. In particular, it will zero out all other
+        //    points in batch related methods such as `batch_normalization`, if even one of the points is the point at infinity.
+        // - This is an optimization that will allow us to do less work, since any scalar multiplied by the point at infinity is
+        //    essentially a no-op.
+        if encoded_g1_element.iter().all(|i| *i == 0) {
             continue;
         }
```

Comment:
This may no longer be needed for safety, since we don't convert from affine to jacobian and then back to affine again, making it strictly an optimization. 

I need to check the blst code and check to see if tests fail to confirm